src/test/java/com/kodilla/ecommercee/ProductTests.java

```diff
@@ -0,0 +1,136 @@
+
+
+    @Test
+    public void testSaveProduct() {
+
+        //Given
+        Product product = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+
+        //When
+        genericEntityRepository.save(product);
+
+        //Then
+        Long id = product.getId();
+        Product foundProduct = (Product) genericEntityRepository.getOne(id);
+        Assert.assertNotEquals(null, foundProduct);
+
+        //CleanUp
+        genericEntityRepository.deleteById(id);
+
+    }
+
+    @Test
+    public void testUpdateProduct() {
+
+        //Given
+        Product product = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+        genericEntityRepository.save(product);
+
+        //When
+        Long id = product.getId();
+        Product productToUpdate = (Product) genericEntityRepository.getOne(id);
+        productToUpdate.setPrice(1000.73);
+        genericEntityRepository.save(productToUpdate);
+
+        //Then
+        productToUpdate = (Product) genericEntityRepository.getOne(id);
+        double updatedPrice = productToUpdate.getPrice();
+        Assert.assertEquals(1000.73, updatedPrice, 0.00);
+
+        //CleanUp
+        genericEntityRepository.deleteById(id);
+
+    }
+
+    @Test
+    public void deleteProduct() {
+
+        //Given
+        Product product1 = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+        Product product2 = new Product("Krzesło Romea", "Krzesło składane z drewna ...", 66.90);
+        Product product3 = new Product("Wkrętarka udarowa", "Wkrętarka marki Hilti ...", 3980.90);
+        genericEntityRepository.save(product1);
+        genericEntityRepository.save(product2);
+        genericEntityRepository.save(product3);
+
+        //When
+        long productCounterBeforeDeletion = genericEntityRepository.count();
+        genericEntityRepository.delete(product1);
+        long productCounter = genericEntityRepository.count();
+
+
+        //Then
+        Assert.assertEquals(productCounterBeforeDeletion-1, productCounter);
+
+        //CleanUp
+        genericEntityRepository.delete(product2);
+        genericEntityRepository.delete(product3);
+    }
+
+    @Test
+    public void getProduct() {
+
+        //Given
+        Product product = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+        genericEntityRepository.save(product);
+
+        //When
+        Long id = product.getId();
+        Product foundProduct = (Product) genericEntityRepository.getOne(id);
+        String productName = foundProduct.getName();
+        String productDescription = foundProduct.getDescription();
+        double productPrice = foundProduct.getPrice();
+
+        //Then
+        Assert.assertEquals("Altana ogrodowa ...", productDescription);
+        Assert.assertEquals("Altana SunSet", productName);
+        Assert.assertEquals(999.90, productPrice, 0.00);
+
+        //CleanUp
+        genericEntityRepository.deleteById(id);
```

Comment:
chyba tak samo co do tej sekcji clean up jak poprzednio :)

Reply:
@aleksandra4991 W JDP190502-18 Paweł napisał, że @Transactional powoduje, że dane testowe zostaną usunięte, wiec usune CleanUp we wszystkich testach.